File: api/src/main/java/com/alten/remotesync/application/notification/record/response/NotificationDTO.java
package com.alten.remotesync.application.notification.record.response;

import com.alten.remotesync.domain.notification.enumeration.NotificationStatus;

import java.time.LocalDateTime;
import java.util.UUID;

public record NotificationDTO(
        UUID notificationId,
        String title,
        String description,
        NotificationStatus status,
        LocalDateTime createdAt
) {
}

File: app/src/app/dto/notification.dto.ts
import { NotificationStatus } from '../dto/notification-status.enum';

export interface NotificationDTO {
  notificationId: string;
  title: string;
  isRead :boolean,
  description: string;
  status: NotificationStatus;
  createdAt: string;
}

export interface PagedNotificationDTO {
  notificationDTOs: NotificationDTO[];
  totalPages: number;
  totalElements: number;
  currentPage: number;
  pageSize: number;
}
export interface InitialNotificationDTO {
  notifications: PagedNotificationDTO;
  urgentCount: number,
  importantCount: number,
  normalCount: number
}
export interface PanelNotificationDTO {
  notifications: PagedNotificationDTO;
  countUnreadNotifications :number
}